drift/lib/src/runtime/cancellation_zone.dart

```diff
@@ -50,8 +50,7 @@ class CancellationToken<T> {
 
 /// Extensions that can be used on cancellable operations if they return a non-
 /// nullable value.
-extension NonNullableCancellationExtension<T extends Object>
-    on CancellationToken<T> {
+extension NonNullableCancellationExtension<T> on CancellationToken<T> {
```

Comment:
This should still have `extends Object` as bounds, otherwise the result of `resultOrNullIfCancelled` would be ambiguous.

The rest of this fix looks good to me, I've pushed a fix making everything non-null. I'll do a final review tomorrow and merge if it looks good.